src/Microsoft.Health.Fhir.SpecManager/Language/TypeScript.cs

```diff
@@ -475,25 +475,30 @@ private void WriteExpandedResourceInterfaceBinding()
                 exportName = complex.NameForExport(FhirTypeBase.NamingConvention.PascalCase, true);
                 string typeName = complex.TypeForExport(FhirTypeBase.NamingConvention.PascalCase, _primitiveTypeMap, false);
 
-                _writer.WriteLineIndented($"export interface {exportName} extends {typeName} {{");
+                if (ShouldSupportGenerics(exportName)) {
+                    _writer.WriteLineIndented($"export interface {exportName}<T extends {typeName} = Resource> {{");
+                } else {
+                    _writer.WriteLineIndented($"export interface {exportName} extends {typeName} {{");
+                }
             }
 
             _writer.IncreaseIndent();
 
             if (isResource)
             {
+                string resourceType;
                 if (ShouldWriteResourceName(complex.Name))
                 {
                     _exportedResources.Add(exportName);
-
-                    _writer.WriteLineIndented("/** Resource Type Name (for serialization) */");
-                    _writer.WriteLineIndented($"resourceType: '{complex.Name}';");
+                    resourceType = complex.Name;
                 }
                 else
                 {
-                    _writer.WriteLineIndented("/** Resource Type Name (for serialization) */");
-                    _writer.WriteLineIndented($"resourceType: string;");
+                    resourceType = "string";
                 }
+
+                _writer.WriteLineIndented("/** Resource Type Name (for serialization) */");
+                _writer.WriteLineIndented($"resourceType: '{resourceType}';");
```

Comment:
Slight error here; this change results in:
```
  resourceType: 'string';
```
instead of:
```
  resourceType: string;
```

I'll push a commit to update this shortly, but in general it is good practice to run at least the TypeScript_R4 (or whatever version you are most comfortable with) to check against the `/generated/TypeScript_R4.ts`.  It is then easy to check correctness and look for unintended changes.